File: iOS/Source/model/Impl/TRTCLiveRoomDef.m
//
//  TRTCLiveRoomDef.m
//  TRTCVoiceRoomOCDemo
//
//  Created by abyyxwang on 2020/7/7.
//  Copyright © 2020 tencent. All rights reserved.
//

#import "TRTCLiveRoomDef.h"

NSString * const Signal_Business_ID = @"businessID";
NSString * const Signal_Business_Live = @"Live";
NSString * const Signal_Platform = @"platform";
NSString * const Signal_Platform_OS = @"iOS";

@implementation TRTCCreateRoomParam

- (instancetype)initWithRoomName:(NSString *)roomName coverUrl:(NSString *)coverUrl{
    self = [super init];
    if (self) {
        self.roomName = roomName;
        self.coverUrl = coverUrl;
    }
    return self;
}

@end

@implementation TRTCLiveRoomConfig

- (instancetype)initWithUseCDNFirst:(BOOL)userCDNFirst cdnPlayDomain:(NSString *)cdnPlayDomain {
    self = [super init];
    if (self) {
        self.useCDNFirst = userCDNFirst;
        self.cdnPlayDomain = cdnPlayDomain;
    }
    return self;
}

@end

@implementation TRTCLiveRoomInfo

- (instancetype)init {
    if (self = [super init]) {
        self.roomId = @"";
        self.roomName = @"";
        self.coverUrl = @"";
        self.ownerId = @"";
        self.ownerName = @"";
        self.streamUrl = @"";
        self.memberCount = 0;
        self.roomStatus = TRTCLiveRoomLiveStatusNone;
    }
    return self;
}

- (instancetype)initWithRoomId:(NSString *)roomId
                      roomName:(NSString *)roomName
                      coverUrl:(NSString *)coverUrl
                       ownerId:(NSString *)ownerId
                     ownerName:(NSString *)ownerName
                     streamUrl:(NSString *)streamUrl
                   memberCount:(NSInteger)memberCount
                    roomStatus:(TRTCLiveRoomLiveStatus)roomStatus{
    self = [super init];
    if (self) {
        self.roomId = roomId;
        self.roomName = roomName;
        self.coverUrl = coverUrl;
        self.ownerId = ownerId;
        self.ownerName = ownerName;
        self.streamUrl = streamUrl;
        self.memberCount = memberCount;
        self.roomStatus = roomStatus;
    }
    return self;
}

@end

@implementation TRTCLiveUserInfo

- (instancetype)initWithProfile:(V2TIMGroupMemberFullInfo *)profile {
    self = [super init];
    if (self) {
        self.userId = profile.userID;
        self.userName = profile.nickName ?: @"";
        self.avatarURL = profile.faceURL ?: @"";
        self.isOwner = profile.role == V2TIM_GROUP_MEMBER_ROLE_SUPER;
    }
    return self;
}

-(NSString *)description {
    return [NSString stringWithFormat:@"ID: %@, Name: %@", self.userId, self.userName];
}

@end

File: Android/tuiliveroom/src/main/java/com/tencent/liteav/liveroom/model/TRTCLiveRoomDef.java
package com.tencent.liteav.liveroom.model;

public class TRTCLiveRoomDef {
    public static final int ROOM_STATUS_NONE     = 0;
    public static final int ROOM_STATUS_SINGLE   = 1;
    public static final int ROOM_STATUS_LINK_MIC = 2;
    public static final int ROOM_STATUS_PK       = 3;


    public static final class TRTCLiveRoomConfig {
        public boolean useCDNFirst;
        public String  cdnPlayDomain;

        public TRTCLiveRoomConfig(boolean useCDNFirst, String cdnPlayDomain) {
            this.useCDNFirst = useCDNFirst;
            this.cdnPlayDomain = cdnPlayDomain;
        }

        @Override
        public String toString() {
            return "TRTCLiveRoomConfig{"
                    + "useCDNFirst=" + useCDNFirst
                    + ", cdnPlayDomain='" + cdnPlayDomain + '\''
                    + '}';
        }
    }


    public static final class TRTCLiveUserInfo {
        public String userId;
        public String userName;
        public String userAvatar;

        @Override
        public String toString() {
            return "TRTCLiveUserInfo{"
                    + "userId='" + userId + '\''
                    + ", userName='" + userName + '\''
                    + ", userAvatar='" + userAvatar + '\''
                    + '}';
        }
    }


    public static final class TRTCCreateRoomParam {
        public String roomName;
        public String coverUrl;

        @Override
        public String toString() {
            return "TRTCCreateRoomParam{"
                    + "roomName='" + roomName + '\''
                    + ", coverUrl='" + coverUrl + '\''
                    + '}';
        }
    }


    public static final class TRTCLiveRoomInfo {
        public int    roomId;
        public String roomName;
        public String coverUrl;
        public String ownerId;
        public String ownerName;
        public String streamUrl;
        public int    roomStatus;
        public int    memberCount;

        @Override
        public String toString() {
            return "TRTCLiveRoomInfo{"
                    + "roomId=" + roomId
                    + ", roomName='" + roomName + '\''
                    + ", coverUrl='" + coverUrl + '\''
                    + ", ownerId='" + ownerId + '\''
                    + ", ownerName='" + ownerName + '\''
                    + ", streamUrl='" + streamUrl + '\''
                    + ", roomStatus=" + roomStatus
                    + ", memberCount=" + memberCount
                    + '}';
        }
    }
}